routers/common/repo.go

```diff
@@ -38,49 +40,75 @@ func ServeBlob(ctx *context.Context, blob *git.Blob, lastModified time.Time) err
 		}
 	}()
 
-	return ServeData(ctx, ctx.Repo.TreePath, blob.Size(), dataRc)
-}
-
-// ServeData download file from io.Reader
-func ServeData(ctx *context.Context, name string, size int64, reader io.Reader) error {
 	buf := make([]byte, 1024)
-	n, err := util.ReadAtMost(reader, buf)
+	n, err := util.ReadAtMost(dataRc, buf)
 	if err != nil {
 		return err
 	}
 	if n >= 0 {
 		buf = buf[:n]
 	}
 
-	ctx.Resp.Header().Set("Cache-Control", "public,max-age=86400")
-
+	size := blob.Size()
 	if size >= 0 {
-		ctx.Resp.Header().Set("Content-Length", fmt.Sprintf("%d", size))
+		ctx.Resp.Header().Set("Content-Length", strconv.FormatInt(size, 10))
 	} else {
-		log.Error("ServeData called to serve data: %s with size < 0: %d", name, size)
+		log.Error("ServeData called to serve data: %s with size < 0: %d", ctx.Repo.TreePath, size)
+	}
+
+	if err := setCommonHeaders(ctx, ctx.Repo.TreePath, buf); err != nil {
+		return err
+	}
+
+	_, err = ctx.Resp.Write(buf)
+	if err != nil {
+		return err
 	}
-	name = path.Base(name)
+	_, err = io.Copy(ctx.Resp, dataRc)
+	return err
+}
 
+func setCommonHeaders(ctx *context.Context, name string, data interface{}) error {
```

Comment:
This does a lot more than setting headers, would suggest removing this function. Cache header can be set like in https://github.com/go-gitea/gitea/pull/20484 via `httpcache` module.